shared/config/certification-settings.ts

```diff
@@ -255,4 +255,30 @@ export const certTypeTitleMap = {
     'JavaScript Algorithms and Data Structures (Beta)'
 };
 
+export type CertSlug = (typeof Certification)[keyof typeof Certification];
+
+export const linkedInCredentialIds = {
```

Comment:
Instead of us inevitably forgetting to add a new one, why not do this pragmatically?

That is, make it some fixed permutation of the cert slug (which _never_ changes).

Reply:
Also, usernames can be changed, so they're not good ids.

Reply:
I am not sure we can get away with that, considering the username is already in the url to the certification?

Reply:
> make it some fixed permutation of the cert slug...

I'm not against that, but it could potentially lead to conflicts. Quincy suggested to use the abbreviation basically. So we could take the first letter and the first letter after each dash - e.g. `data-visualization` -> `dv`, and `front-end-development-libraries` -> `fedl` - but `javascript-algorithms-and-data-structures-v8` would be `jaadsv`, which would be the same for `javascript-algorithms-and-data-structures-v99`. I suppose we could add the whole v# - e.g. Maybe `if (/^v\d+$/.test(lastWord)) -> add to id`.

So I guess that would work. I would be fine with it if we want to do that. Otherwise, we could add a test to make sure all the certs have an id. What do you think?

Some info after looking into this... I found [this article about linkedin certifications](https://cufinder.io/blog/what-is-credential-id-in-linkedin-certification/) - basically, the credential ID is something created by the issuing organization. Here's [what a Microsoft cert looks like](https://wedoazure.files.wordpress.com/2023/07/image-3.png?w=1024) - they give, and presumably store, a unique ID for each certification issued. We don't really have anything like that, we're just making up a string - which is fine if that's what we want to do, but it doesn't really feel like we're doing it right. Perhaps we want to start storing some sort of ID like that.

Reply:
> Perhaps we want to start storing some sort of ID like that.

I think so, yeah. That way we can ensure uniqueness and that it's tied to a particular user account. A uuid?

Reply:
So thinking through what that would all take...

We would need to store the credential ID in the database, presumably on the user object:
1: Add it to the completedChallenges array, so you might get a completed cert challenge that looks like this:

```
{
  id: '660add10cb82ac38a17513be',
  completedDate: 1515450994243,
  solution: null,
  challengeType: 7,
  credentialId: uuid,
  files: []
}
```

or 2: create a new place for it - something like:

```
certificateCredentialIds: {
  Certifications.RespWebDesign: uuid
}
```

I initially thought it's own area would be better, but I kind of like in the completedChallenges cause we could add a credential ID for anything we want there. Maybe we want to add a unique ID for certification projects or something in the future. Something to discuss.

We would add this credential ID when users claim a cert - but users wouldn't have it on previously claimed certs (already in the database). So we could do a database migration to add it to all the user records, or maybe create it when they try to access the cert or something. Both might be a little tricky, but doable. Or just leave previously claimed certs without that ID, but I don't really like that.

Reply:
> Or just leave previously claimed certs without that ID, but I don't really like that.

I think it's fine to have the ID creation happen when they add the certification to LinkedIn. Otherwise, what is it referring to?

Also, if the id only exists if they've clicked the button, we can use that fact to see how many people have added their certifications to LinkedIn.

Reply:
I don't think we're done discussing this. Using the username is a mistake because usernames can change.

user 'abc' claims a certification 'abc-dv', user 'abc' changes their username to '123', new user gets username 'abc'. Now there's a problem. They either cannot claim the certification on LinkedIn or they already have it. I'm not sure, but, regardless we've violated the uniqueness constraint (from LinkedIn's perspective).

Reply:
@ojeytonwilliams I do not understand the issue. We already encode `username` in the linkedin URL to the certification. Currently (pre this PR), if a Camper changes their username, their link on linkedin becomes invalid.

Reply:
As I understand it, the `certId` must be unique, not the `certUrl`. I.e. that's a requirement from LinkedIn.

I agree that certification urls should be permanent, but that's a bit out of scope.

Reply:
And to confirm: globally unique?

Reply:
I thought so, but Tom's been investigating and it seems not :/